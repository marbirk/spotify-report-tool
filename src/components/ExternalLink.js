import React from 'react'

const ExternalLink = ({ children, ...props }) => (
    <a target="_blank" rel="noreferrer noopener" {...props}>
        {children}
    </a>
)

export default ExternalLink
