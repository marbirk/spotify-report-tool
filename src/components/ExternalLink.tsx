import React from 'react'

interface ExternalLinkProps {
    children: React.ReactNode
    className?: string
    href: string
}

const ExternalLink = ({ children, ...props }: ExternalLinkProps) => (
    <a target="_blank" rel="noreferrer noopener" {...props}>
        {children}
    </a>
)

export default ExternalLink
