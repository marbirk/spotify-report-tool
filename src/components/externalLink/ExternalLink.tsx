import React from 'react'

interface ExternalLinkProps {
    children: React.ReactNode
    className?: string
    href: string
}

const ExternalLink = ({ children, ...other }: ExternalLinkProps) => (
    <a target="_blank" rel="noreferrer noopener" {...other}>
        {children}
    </a>
)

export default ExternalLink
