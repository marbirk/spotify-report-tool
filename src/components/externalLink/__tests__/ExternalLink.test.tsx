import React from 'react'
import { render } from '@testing-library/react'

import ExternalLink from '../ExternalLink'

describe('ExternalLink', () => {
    test('renders correctly', () => {
        const { container, getByText } = render(
            <ExternalLink href="https://test-url.de">
                External Link
            </ExternalLink>
        )
        expect(getByText('External Link')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})
