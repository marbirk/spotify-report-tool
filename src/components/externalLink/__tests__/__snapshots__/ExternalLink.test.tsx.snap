// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders correctly 1`] = `
<div>
  <a
    href="https://test-url.de"
    rel="noreferrer noopener"
    target="_blank"
  >
    External Link
  </a>
</div>
`;
